### PR TITLE
Remove --json, add --mode. Pass in terminals to error handlers

### PR DESCRIPTION
**Pull request by fredemmott**
Prep work for adding LSP support.

@PranayAgarwal  : I'd add you as a reviewer if I could :)

### CODE REVIEW

File: src/__Private/LinterCLI.php [hunk 1]
```diff
@@ -169,12 +173,14 @@ private function lintPath(
...
-        'is_interactive' => $this->isInteractive(),
-      ), $this->getStdout(), $this->getStderr());
+    switch ($this->mode) {
+      case LinterCLIMode::PLAIN:
+        $error_handler = new LinterCLIErrorHandlerPlain($this->getTerminal());
+        break;
+      case LinterCLIMode::JSON:
+        $error_handler = new LinterCLIErrorHandlerJSON($this->getTerminal());
+        break;
+    }
```
[the remark this answers is not in the document]
Comment (fredemmott) [reply src/__Private/LinterCLI.php]: yep, it knows the switch is exhaustive

**Comment by PranayAgarwal** [comment]
Looks like some tests are failing

**Comment by fredemmott** [comment]
yeah, test is still passing in --json

File: src/__Private/LinterCLI.php [hunk 2]
```diff
@@ -57,6 +57,11 @@ protected function getSupportedOptions(): vec<CLIOptions\CLIOption> {
         '--mode',
         '-m',
       ),
+      CLIOptions\flag(
+        () ==> { $this->mode = LinterCLIMode::JSON; },
+        'Equivalent to --mode=json',
+        '--json'
+      ),
       $this->getVerbosityOption(),
```
Comment (PranayAgarwal) [inline src/__Private/LinterCLI.php]: Why re-add `--json` vs. updating tests?

**Comment by fredemmott** [comment]
Consistency with hh_client - but just realized I won’t do that anyway, as hh_cloent has no dashes before ‘lsp’

**Pull request merged**